mm/2s2h/Rando/Logic/Regions/Central.cpp

```diff
@@ -197,7 +202,8 @@ static RegisterShipInitFunc initFunc([]() {
     };
     Regions[RR_CURIOSITY_SHOP_FRONT] = RandoRegion{ .name = "Front", .sceneId = SCENE_AYASHIISHOP,
         .checks = {
-            // TODO : Add the shop checks
+            CHECK(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM, CAN_AFFORD(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_RECOVERED_STOLEN_BOMB_BAG)),
```

Comment:
I'm surprised this one is passing, but you should remove the unset flag check here, the logic doesn't remove flags unless the scene is no longer in logic currently.
```suggestion
            CHECK(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM, CAN_AFFORD(RC_BOMB_SHOP_ITEM_OR_CURIOSITY_SHOP_ITEM)),
```